vanity/file.go

```diff
@@ -29,14 +29,16 @@
 package vanity
 
 import (
+	"strings"
+
 	"github.com/gogo/protobuf/gogoproto"
 	"github.com/gogo/protobuf/proto"
 	descriptor "github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
-	"strings"
 )
 
-func NotInPackageGoogleProtobuf(file *descriptor.FileDescriptorProto) bool {
-	return !strings.HasPrefix(file.GetPackage(), "google.protobuf")
+func NotGoogleProtobufDescriptorProto(file *descriptor.FileDescriptorProto) bool {
+	// can not just check if file.GetName() == "google/protobuf/descriptor.proto" because we do not want to assume compile path
+	return !(file.GetPackage() == "google.protobuf" && strings.HasSuffix(file.GetName(), "descriptor.proto"))
```

Comment:
You're right, you could - this isn't perfect, it's a simple heuristic. The most exact solution we could have would be to store the FileDescriptorProto outputted for descriptor.proto as a global variable, and then compare it to the parameter. However, I would argue this is potentially a worse heuristic, as when descriptor.proto changes, the comparison will fail.